R/rd-data.R

```diff
@@ -40,6 +40,52 @@ as_data.tag_item <- function(x, ...) {
 parse_section <- function(x, title, ...) {
   text <- flatten_para(x, ...)
 
+  html <- xml2::read_html(text)
+  r_blocks <- xml2::xml_find_all(html, "//div[contains(@class, 'sourceCode r')]/pre/code")
+
+  highlight_r_block <- function(block) {
+    xml2::xml_text(block) <- downlit::highlight(
+      xml_text1(block),
+      classes = downlit::classes_pandoc()
+    )
+  }
+
+  purrr::walk(r_blocks, highlight_r_block)
+
+  non_r_blocks <- xml2::xml_find_all(
+    html,
+    "//div[contains(@class, 'sourceCode') and not(contains(@class, 'sourceCode r'))]"
+  )
+
+  highlight_other_block <- function(block) {
+    lang <- sub("sourceCode ", "", xml2::xml_attr(block, "class"))
+    code <- xml2::xml_text(xml2::xml_find_first(block, "pre/code"))
+    highlighted <- markdown_text(
```

Comment:
a new use case for `markdown_text()` :grin: In that function `pkg` is needed for finding external links but in code we do not need to try and identify them I think (and passing `pkg` all the way to here would warrant more refactoring).